File: solutions/python/darts/1/darts.py
import math

def score(x, y):
    radius = math.sqrt(x**2 + y**2)

    if 0 <= radius <= 1:
        return 10
    if 1 < radius <= 5:
        return 5
    if 5 < radius <= 10:
        return 1
    return 0

File: solutions/ruby/darts/1/darts.rb
class Darts

  TP = TARGET_POINTS = {
        ..1  => 10,
     1.1..5  => 5,
     5.1..10 => 1,
    10.1..   => 0
  }

  private_constant :TP

  private

  def initialize(x, y)
    @gipotenuza = Math.sqrt(x**2 + y**2).ceil
  end

  public

  def score
    TP.select { |range, point| range.cover?(@gipotenuza) }.values.first || 0
  end

end